include/multipass/constants.h

```diff
@@ -28,6 +28,8 @@ namespace multipass
 constexpr auto client_name = "multipass";
 constexpr auto daemon_name = "multipassd";
 
+constexpr auto snapcraft_remote = "snapcraft";
+
```

Comment:
Two doubts here:
1. There other three remotes "release", "daily", "appliance" strings are defined in ubuntu_image_host.h, but "snapcraft" is defined in constants.h, is there a reason behind that? 
2. I know that the codebase has been using auto to catch string literal, which makes the type char*. What about C++17 feature std::string_view, it might result in some client code changes, but it has its upsides, like more public methods and better performance (has length info, no longer a null-terminated which requires traversal for every operation).

Reply:
Hey @georgeliao!

1. I want to start moving those constants to, well, `constants.h`, so since I'm modifying things for the `snapcraft` remote in this PR, this is a good place to start.  I'm reluctant to change the other remotes in this PR since that wouldn't be really part of what this PR is accomplishing.
2. I think this is a good idea in changing these string literals to be `string_view` instead of `auto`, but I don't think this is the PR to do it.  I *could* make this particular change be that though.

Reply:
Ah, wait, yes, I'm going to revert the `string_view` change since it's breaking the other remotes.  I think if we want to change string literal constants to `string_view`, that should be a separate PR with all of the changes necessary in the code.

Reply:
I think the client code for snapcraft_remote variable has to change accordingly, because char* can be converted to std::string implicitly, but std::string_view can not, because the `explicit std::string(std::string_view)` constructor is explicit. So std::string(snapcraft_remote) needs to be called to fix the compilation error. Anyway, it is a minor thing though, it is also ok to skip this for now.